examples/functions/stale-products/index.ts

```diff
@@ -0,0 +1,217 @@
+import {documentEventHandler} from '@sanity/functions'
+import {createClient} from '@sanity/client'
+
+interface PagePayload {
+  _id: string
+  _type: string
+  modules?: Array<{
+    _type: string
+    items?: Array<{
+      _type: string
+      productWithVariant?: {
+        product?: {
+          _id: string
+        }
+      }
+    }>
+  }>
+}
+
+interface ProductWithDates {
+  _id: string
+  _createdAt: string
+  _updatedAt: string
+  title?: string
+  store?: {
+    title?: string
+  }
+}
+
+const STALE_PRODUCT_THRESHOLD_DAYS = 30
+
+export const handler = documentEventHandler(async ({context, event}) => {
+  console.log('📄 Page Product Age Analysis Function called at', new Date().toISOString())
+  console.log('📝 Event:', event)
+
+  try {
+    const {_id, _type} = event.data as PagePayload
+
+    if (_type !== 'page') {
+      console.log('⏭️ Skipping non-page document:', _type)
+      return
+    }
+
+    const client = createClient({
+      ...context.clientOptions,
+      useCdn: false,
+      apiVersion: '2025-06-01',
+    })
+
+    console.log('🔍 Analyzing product ages for page:', _id)
+
+    // Fetch the page with its modules and product data in one query
+    // We're intentionally showing a more complex query to demonstrate real world usage,
+    // you may have additional modules/fields with product references you'd want to analyze
+    const page = await client.fetch<{
+      _id: string
+      modules: Array<{
+        _type: string
+        items?: Array<{
+          _type: string
+          productWithVariant?: {
+            product?: {
+              _id: string
+              _createdAt: string
+              _updatedAt: string
+              title?: string
+              store?: {
+                title?: string
+              }
+            }
+          }
+        }>
+      }>
+    }>(
+      `*[_id == $id][0] {
```

Comment:
also prefer assigning queries to a variable at the top

Reply:
moved this to a top query